packages/quantic/force-app/main/default/lwc/quanticRefineModalContent/quanticRefineModalContent.js

```diff
@@ -59,9 +59,15 @@ export default class QuanticRefineModalContent extends LightningElement {
   hasActiveFilters = false;
   /** @type {AnyHeadless} */
   headless;
+  /** @type {object} */
+  renderedFacets = {};
+  /** @type {boolean} */
+  areFacetsRendered = false;
+
 
   connectedCallback() {
     registerComponentForInit(this, this.engineId);
+    this.addEventListener('renderFacet', this.handleRenderFacetEvent);
```

Comment:
It would be nice to remove the event listener in `disconnectedCallback`.

Reply:
absolutely!